app/src/modules/fota/fota.c

```diff
@@ -130,59 +159,97 @@ static void state_initialized_run(void *o)
 		const enum cloud_status status = MSG_TO_CLOUD_STATUS(state_object->msg_buf);
 
 		if (status == CLOUD_CONNECTED_READY_TO_SEND) {
-			STATE_SET(STATE_POLL_AND_PROCESS);
+			STATE_SET(STATE_WAIT_FOR_TRIGGER);
 		}
 	}
 }
 
-static void state_poll_and_process_run(void *o)
+static void state_wait_for_trigger_run(void *o)
 {
 	struct s_object *state_object = o;
 
 	if (&TRIGGER_CHAN == state_object->chan) {
-		int err;
 		const enum trigger_type trigger_type = MSG_TO_TRIGGER_TYPE(state_object->msg_buf);
-		enum fota_status fota_status = FOTA_STATUS_PROCESSING_START;
 
-		if (trigger_type != TRIGGER_FOTA_POLL) {
-			return;
+		if (trigger_type == TRIGGER_FOTA_POLL) {
+			STATE_SET(STATE_POLL_AND_PROCESS);
 		}
+	}
 
-		/* Tell the rest of the app not to disturb */
-		err = zbus_chan_pub(&FOTA_STATUS_CHAN, &fota_status, K_NO_WAIT);
-		if (err) {
-			LOG_ERR("zbus_chan_pub, error: %d", err);
-			SEND_FATAL_ERROR();
+	if (&CLOUD_CHAN == state_object->chan) {
+		const enum cloud_status status = MSG_TO_CLOUD_STATUS(state_object->msg_buf);
+
+		if (status == CLOUD_CONNECTED_PAUSED) {
```

Comment:
Not sure I follow the logic here, we go to initialized state whenever the cloud connection is paused?

Reply:
If cloud connection is paused, which only happens if network connection is lost and a trigger would cause a failed FOTA status request, we go to the initialized state, yes. In the initialized state, the module waits for `CLOUD_CONNECTED_READY_TO_SEND` and then goes back to waiting for triggers.

Reply:
Alright, I'm mostly used to the word "initialization" being used for at boot one time actions.

Reply:
Yeah, maybe `STATE_PAUSED` would be a better name of the state

Reply:
Changed it to `STATE_WAIT_FOR_CLOUD`, it's more descriptive